File: src/main/js/my-plant-diary-gui/src/app/api/models/create-event-so.ts
/* tslint:disable */
/* eslint-disable */
import { EventTypeSo } from './event-type-so';
import { PlantBasicSo } from './plant-basic-so';
export interface CreateEventSo {
  dateTime: string;
  description?: string;
  plant: PlantBasicSo;
  type: EventTypeSo;
}

File: src/main/java/sk/filo/plantdiary/service/so/CreateEventSO.java
package sk.filo.plantdiary.service.so;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;

@Schema(name = "CreateEvent")
@Getter
@Setter
@ToString
public class CreateEventSO {

    @NotNull
    @Valid
    private EventTypeSO type;

    @NotNull
    private LocalDateTime dateTime;

    @Size(max = 1000)
    private String description;

    @NotNull
    @Valid
    private PlantBasicSO plant;

}
